src/System.Private.CoreLib/shared/System/Resources/FastResourceComparer.cs

```diff
@@ -32,8 +33,9 @@ public int GetHashCode(object key)
             return FastResourceComparer.HashFunction(s);
         }
 
-        public int GetHashCode(string key)
+        public int GetHashCode(string? key) // TODO-NULLABLE: argument should be non-nullable but IEqualityComparer.Equals accepts null
```

Comment:
Also, we could potentially just disable the warning here if we want this to not accept null. We could just use pragma to disable the error, no?

Reply:
I think we need to figure out how this should work correctly with generics, once we have the answer we should fix it up or remove TODOs, either way we will need a second pass on the code

Reply:
I think this should be fixed once the compiler enables constraining or widening parameters on overriding members. That is why I was suggesting disabling the warning.

Reply:
@safern I'm not sure constraining would be a right fix here since consumer sees nullable on the interface and then non-nullable on the class, this will be confusing. I'm rather leaning toward fixing implementation to work with null but got no strong opinion

Reply:
I think it is fine to leave as is with the TODO comment specially since it is an internal type. I thought we just wanted to annotate the intent.